rsp_jupyter_extensions/handlers/query.py

```diff
@@ -1,112 +1,288 @@
         # URL
         params = {"query_url": url}
 
-        # Get the endpoint for the rendered URL
-        obj = self._client.get(nb_url).json()
-        rendered_url = obj["rendered_url"]
+        return self._get_nublado_seeds_notebook(
+            notebook,
+            params
+        )
 
-        # Retrieve that URL and return the textual response, which is the
-        # string representing the rendered notebook "in unicode", which I
-        # think means, "a string represented in the default encoding".
-        return self._client.get(rendered_url, params=params).text
+    def _get_query_all_notebook(self) -> str:
+        """Even-more-curried helper function for TAP history notebook."""
+        notebook = "history"
+        params: dict[str,str] = {}
+        return self._get_nublado_seeds_notebook(
+            notebook,
+            params
+        )
+
+
+
+    @tornado.web.authenticated
+    async def get(self, *args: str, **kwargs: str) -> None:
+        #
+        # The only supported querytype for now is "tap"
+        #
+        # GET .../<qtype>/<id> will act as if we'd posted a query with
+        #     qytpe and id
+        # GET .../<qtype>/history/<n> will request the last n queries of
+        #     that type.
+        # GET .../<qtype>/notebooks/query_all will create and open a notebook
+        #     that will ask for all queries and yield their jobids.
+
+        path = self.request.path
+        stem = "/rubin/query"
+
+        route = _peel_route(path, stem)
+        if route is None:
+            self.log.warning(f"Cannot strip '{stem}' from '{path}'")
+            raise UnimplementedQueryResolutionError(path)
+        route = route.strip("/")  # Remove leading and trailing slashes.
+        components = route.split("/")
+        if len(components) < 2 or len(components) > 3:
+            self.log.warning(
+                f"Cannot parse query from '{path}' components '{components}'"
+            )
+            raise UnimplementedQueryResolutionError(path)
+        q_type = components[0]
+        match q_type:
+            case "tap":
+                await self._tap_route_get(components[1:])
+            case _:
+                raise UnsupportedQueryTypeError(
+                    f"{q_type} is not a supported query type"
+                )
+
+    async def _tap_route_get(self, components: list[str]) -> None:
+        if components[0] == "history":
+            if len(components) == 1:
+                self.write(await self._generate_query_all_notebook())
+                return
+            s_count = components[1]
+            try:
+                count = int(s_count)
+            except ValueError as exc:
+                raise UnimplementedQueryResolutionError(
+                    f"{self.request.path} -> {exc!s}"
+                ) from exc
+            jobs = await get_query_history(count)
+            qtext = self._get_query_text(jobs)
+            q_dicts = [ x.dict() for x in qtext ]
```

Comment:
Should we replace dict() since I think it's deprecated for Pydantic?
```suggestion
            q_dicts = [x.model_dump() for x in qtext]
```